extensions/wikia/PortableInfobox/tests/PortableInfoboxMobileRenderServiceTest.php

```diff
@@ -0,0 +1,609 @@
+<?php
+
+class PortableInfoboxMobileRenderServiceTest extends WikiaBaseTest {
+	//todo: https://wikia-inc.atlassian.net/browse/DAT-3076
+	//todo: we are testing a lot of functionality and have issues with mocking
+	//todo: we should move all render service test to API tests
+
+	protected function setUp() {
+		$this->setupFile = dirname( __FILE__ ) . '/../PortableInfobox.setup.php';
+		parent::setUp();
+	}
+
+	private function mockInfoboxRenderServiceHelper( $input ) {
+		$extendImageData = isset( $input['extendImageData'] ) ? $input['extendImageData'] : null;
+
+		$mock = $this->getMockBuilder( 'Wikia\PortableInfobox\Helpers\PortableInfoboxImagesHelper' )
+			->setMethods( [ 'extendImageData' ] )
+			->getMock();
+		$mock->expects( $this->any() )
+			->method( 'extendImageData' )
+			->will( $this->returnValue( $extendImageData ) );
+
+		$this->mockClass( 'Wikia\PortableInfobox\Helpers\PortableInfoboxImagesHelper', $mock );
+	}
+
+	/**
+	 * @param $html
+	 * @return string
+	 */
+	private function normalizeHTML( $html ) {
+		if ( empty( $hrml ) ) {
+			return '';
+		}
+
+		$DOM = new DOMDocument( '1.0' );
+		$DOM->formatOutput = true;
+		$DOM->preserveWhiteSpace = false;
+		$DOM->loadXML( $html );
+
+		return $DOM->saveXML();
+	}
+
+	/**
+	 * @param $input
+	 * @param $expectedOutput
+	 * @param $description
+	 * @param $mockParams
+	 * @dataProvider testRenderInfoboxDataProvider
+	 */
+	public function testRenderInfobox( $input, $expectedOutput, $description, $mockParams ) {
+		$wrapper = new \Wikia\Util\GlobalStateWrapper( [ 'wgArticleAsJson' => $mockParams['isMercury'] ?? true ] );
+		$this->mockInfoboxRenderServiceHelper( $mockParams );
+
+		$infoboxRenderService = new PortableInfoboxMobileRenderService();
+		$actualOutput = $wrapper->wrap( function () use ( $infoboxRenderService, $input ) {
+			return $infoboxRenderService->renderInfobox( $input, '', '', '', '' );
+		} );
+		$expectedHtml = $this->normalizeHTML( $expectedOutput );
+		$actualHtml = $this->normalizeHTML( $actualOutput );
+
+		$this->assertEquals( $expectedHtml, $actualHtml, $description );
+	}
+
+	public function testRenderInfoboxDataProvider() {
+		return [
+			[
+				'input' => [ ],
+				'output' => '',
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "output" 15 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)